### PR TITLE
TSDK-825 Normal operation single

### PR DESCRIPTION
**Pull request by mundacho**
## Purpose

The goal of this PR is to simulate the workings of the PBFT algorithm but with only one node.

## Approach

- We moved the GRPC client to the common project.
- We implemented a new state machine for the replication, this new state machine has a persistence layer.
- We modified the WS to return final states instead of the 404.
- We separated the monitor from the PBFT machinery, and connected it using GRPC.

## Testing

All integration tests that existed before continue to pass

## Tickets

* TSDK-825

### CODE REVIEW

File: consensus/src/main/scala/co/topl/bridge/consensus/modules/ApiServicesModule.scala [hunk 1]
```diff
@@ -186,6 +431,152 @@ trait ApiServicesModule {
                       )
                     case StartSession(sc) =>
                       startSession(request.clientNumber, request.timestamp, sc)
+                    case PostDepositBTC(
+                          value
+                        ) => // FIXME: add checks before executing
```
Comment (DiademShoukralla) [inline consensus/src/main/scala/co/topl/bridge/consensus/modules/ApiServicesModule.scala:436]: These FIXMEs on line 436, 445, 453, etc.. are they meant to be fixed within this PR or was this a note for a future task?

File: consensus/src/main/scala/co/topl/bridge/consensus/monitor/TransitionToEffect.scala [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+            .start(
+              consensusClient.timeoutTBTCMint(
+                TimeoutTBTCMintOperation(
+                  session.id,
+                  ev.height
+                )
+              )
+            )
+            .void
+        case ( // TODO: make sure that by the time we are here, the funds are already locked
```
Comment (DiademShoukralla) [inline consensus/src/main/scala/co/topl/bridge/consensus/monitor/TransitionToEffect.scala:168]: Same question with this TODO and the FIXME on line 201

**Review by DiademShoukralla — APPROVED** [review]
LGTM!

File: consensus/src/main/scala/co/topl/bridge/consensus/modules/ApiServicesModule.scala (hunk 1 above)
Comment (mundacho) [reply consensus/src/main/scala/co/topl/bridge/consensus/modules/ApiServicesModule.scala:436]: This was note for a future task, I will note that next time.

File: consensus/src/main/scala/co/topl/bridge/consensus/monitor/TransitionToEffect.scala (hunk 2 above)
Comment (mundacho) [reply consensus/src/main/scala/co/topl/bridge/consensus/monitor/TransitionToEffect.scala:168]: same thing as before

**Pull request merged**